db/seeds.rb

```diff
@@ -1,50 +1,57 @@
+# frozen_string_literal: true
+
 require 'csv'
 require 'faker'
 
 # add cities in DB
 
-City.all.destroy_all
+City.find_each(&:destroy!)
 
-source_file = "/Users/eleonore/Documents/Code/enm-city-allocator/db/enm_cities_seats.csv"
+source_file = '/Users/eleonore/Documents/Code/enm-city-allocator/db/jurisdictions_and_choices.csv'
 
 CSV.foreach(source_file, headers: true) do |row|
-	City.create(
-		name: row[1],
-		nb_seats: row[2]
-		)
+  City.create!(
+    name: row[0],
+    nb_seats: row[1],
+    wishes: [row[2], row[3], row[4], row[5], row[6], row[7]].map(&:to_i)
+  )
 end
 
 # add students in DB
 
-Student.all.destroy_all
+Student.find_each(&:destroy!)
 
-Student.create(name: "Anna Thuleau")
+Student.create!(name: 'Anna Thuleau')
 
-335.times {
-	Student.create!(
-	name: Faker::Name.unique.name
-	)
-}
+335.times do
+  Student.create!(
+    name: Faker::Name.unique.name
+  )
+end
 
 # add choices in DB
 
-Choice.all.destroy_all
-
-Student.all.each do |student|
-	rank = 1
-	6.times {
-		n = rand(100)
-		if n < 20
-			random_city = City.find_by name: 'PARIS (PI)'
-		elsif n < 40
-			random_city = City.find_by name: 'BORDEAUX (PI)'
-		else
-			random_city = City.all.sample(100)[50]
-		end
-		Choice.create!(
-			city: random_city, 
-			student: student,
-			rank: rank)
-		rank += 1
-	}
-end
\ No newline at end of file
+Choice.find_each(&:destroy!)
+
+City.order("nb_seats DESC").all.each do |city|
+  city.wishes.each_with_index do |wish, ind|
+    # next if wish.zero?
+
+    until wish.zero?
```

Comment:
Since your loop will run exactly `wish` times, it's best to not use a `while`/`until`. Always avoid unbounded loops as much as possible, because they always carry the risk of an infinite loop

Reply:
Ok thanks @Berardpi, so what you suggest is for example to have instead:
`wish.times { # my loop }`
?

Reply:
Also, `wish` is a positive integer and we have a 'decremeting line' that is not part of any condition, so isn't that sure that at some point, `wish` will reach the value 0 (and hence the loop will never be infinite)?

Reply:
I choose this way of doing as I thought this way it was clearer that we remove a wish each time

Reply:
Yes I would have gone with a `times`.

It sure works in this case. My comment was more general. We tend to avoid non-finite loops (while / until) as much as possible and favor finite loop (each, map, times, ...) as much as possible.
This is because a while/until always carries a risk:
- If for any reason wish equal -1, then your loop will never end
- If a future code change introduce an early loop exit (`next`) then it will create a risk